From 65651c8e1391fa356a5cd2c0341fa4e3486d1cd9 Mon Sep 17 00:00:00 2001
From: Matteo Facchinetti <matteo.facchinetti@sirius-es.it>
Date: Wed, 8 Feb 2017 17:09:10 +0100
Subject: [PATCH] backlight: tps65217_bl: disable backlight when system
 shutdown

When system shutdown, it's better disable backlight signal at first
and then turn off the lcd.

Signed-off-by: Matteo Facchinetti <matteo.facchinetti@sirius-es.it>
---
 drivers/video/backlight/tps65217_bl.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/video/backlight/tps65217_bl.c b/drivers/video/backlight/tps65217_bl.c
index c6706f5..be2d879 100644
--- a/drivers/video/backlight/tps65217_bl.c
+++ b/drivers/video/backlight/tps65217_bl.c
@@ -407,9 +407,19 @@ static int tps65217_bl_remove(struct platform_device *pdev)
 	return 0;
 }
 
+static int tps65217_bl_shutdown(struct platform_device *pdev)
+{
+	struct tps65217_bl *tps65217_bl = platform_get_drvdata(pdev);
+
+	tps65217_bl_disable(tps65217_bl);
+
+	return 0;
+}
+
 static struct platform_driver tps65217_bl_driver = {
 	.probe		= tps65217_bl_probe,
 	.remove		= tps65217_bl_remove,
+	.shutdown       = tps65217_bl_shutdown,
 	.driver		= {
 		.owner	= THIS_MODULE,
 		.name	= "tps65217-bl",
-- 
2.7.4

